cli/src/commands/split.rs

```diff
@@ -230,3 +242,26 @@ The remainder will be in the second commit.
     tx.finish(ui, format!("split commit {}", commit.id().hex()))?;
     Ok(())
 }
+
+/// Reads 'split.legacy-bookmark-behavior' from settings and prints a warning if
+/// the value is unset to alert the user to the behavior change.
+fn read_legacy_bookmark_behavior_setting(
+    settings: &jj_lib::settings::UserSettings,
+    ui: &Ui,
+) -> Result<bool, CommandError> {
+    match settings.get_bool("split.legacy-bookmark-behavior") {
+        Err(ConfigGetError::NotFound { name: _ }) => {
+            writeln!(
```

Comment:
I didn't think super hard about the message we should show. Let me know if you have other ideas.

Reply:
We've tweaked this a bit, and I think it's good now.